packages/headless/src/controllers/folded-result-list/headless-folded-result-list.test.ts

```diff
@@ -106,5 +110,81 @@ describe('FoldedResultList', () => {
         'folding/loadCollection/pending'
       );
     });
+
+    it('finds a result by id', () => {
+      const expected = foldedResultList.state.results[2];
+      const result = foldedResultList.findResultById(expected);
+      expect(result).toEqual(expected);
+    });
+
+    it('finds a result nested one level by id', () => {
+      const expected = makeFoldedResult({
+        uniqueId: 'unique_id_to_look_for',
+      });
+      foldedResultList.state.results[2].children.push(expected);
+      const result = foldedResultList.findResultById(expected);
+      expect(result).toEqual(expected);
+    });
+
+    it('finds a result nested two levels by id', () => {
+      const expected = makeFoldedResult({uniqueId: 'unique_id_to_look_for'});
+      const firstChild = makeFoldedResult(
+        {
+          uniqueId: 'first_child',
+        },
+        [expected]
+      );
+      foldedResultList.state.results[2].children.push(firstChild);
+      const result = foldedResultList.findResultById(expected);
+      expect(result).toEqual(expected);
+    });
+
+    it('finds a result by collection', () => {
+      const expected = foldedResultList.state.results[2];
+      const result = foldedResultList.findResultByCollection(expected);
+      expect(result).toEqual(expected);
+    });
+
+    it('finds a result nested one level by collection', () => {
+      const expected = makeFoldedResult({
+        raw: {
+          foldingcollection: 'collection_to_look_for',
+          urihash: '',
+        } as Raw,
+      });
+      foldedResultList.state.results[0].children.push(expected);
+      const result = foldedResultList.findResultByCollection(expected);
+      expect(result).toEqual(expected);
+    });
+
+    it('finds a result nested two levels by collection', () => {
+      const expected = makeFoldedResult({
+        raw: {
+          foldingcollection: 'collection_to_look_for',
+          urihash: '',
+        } as Raw,
+      });
+      const firstChild = makeFoldedResult(
+        {
+          raw: {
+            foldingcollection: 'first_child',
+            urihash: '',
+          } as Raw,
+        },
+        [expected]
+      );
+      foldedResultList.state.results[2].children.push(firstChild);
+      const result = foldedResultList.findResultByCollection(expected);
+      expect(result).toEqual(expected);
+    });
   });
 });
+
+function makeFoldedResult(result: any, children: any[] = []) {
```

Comment:
I would avoid `any`.

```suggestion
function makeFoldedResult(result: Result, children: Result[] = []) {
```

Reply:
oopsie

Reply:
I thought we allowed it in tests! (not that anyone told me we did, I just assumed)